models/generalinfo.js

```diff
@@ -22,7 +22,19 @@ const GeneralinfoSchema = new Schema({
 		type: String,
 		enum: require("../public/dist/js/categories").categories,
 		required: "General information requires a category"
+	},
+	school: {
+		type: Schema.Types.ObjectId,
+		required: "General information requires a school or to be null"
 	}
 });
 
+GeneralinfoSchema.pre("validate", function (next) {
+	const generalinfo = this;
+	if (generalinfo.school == null) {
+		generalinfo.school = require("mongoose").Types.ObjectId("000000000000000000000001");
```

Comment:
What is the reason for using this weird ObjectId instead of keeping it at null? Global general information can easily be  identified by having a school value of null right?

Reply:
That was my initial approach but I was getting an error of converting the null to objectID.

So i chose an easily recognizable objectID to act as our Global id.

Reply:
Hmm, did you removed the `required` attribute?

Reply:
The problem is when the casting to objectID happens null is not a valid objectID. 
I tried both leaving the request as is or specifically turning it into null in the pre hook. Both gave the same error that null is not a valid objectID

Reply:
Hmm weird, it should be possible like that. In any case, if this works then we can keep it like this for now.

Reply:
yeah this works as intended we can revisit it on a later date.